extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/groupby/GroupByQueryKit.java

```diff
@@ -151,25 +150,45 @@ public QueryDefinition makeQueryDefinition(
                        .processorFactory(new GroupByPreShuffleFrameProcessorFactory(queryToRun))
     );
 
+    List<KeyColumn> resultClusterByWithPartitionBoostColumns = new ArrayList<>(resultClusterByWithoutPartitionBoost.getColumns());
+    resultClusterByWithPartitionBoostColumns.add(new KeyColumn(QueryKitUtils.PARTITION_BOOST_COLUMN, KeyOrder.ASCENDING));
+    ClusterBy resultClusterByWithPartitionBoost = new ClusterBy(
+        resultClusterByWithPartitionBoostColumns,
+        resultClusterByWithoutPartitionBoost.getBucketByCount()
```

Comment:
I think line 153 : 167 can be another method called createRowSignature. 
Also the if on line 159 should be the first switch in the control flow. Something like 
```
SigX=null; 
if(boosted)
{
sigX=foo
}else{
sigX=bar
}
```

Reply:
Refactored to different function, and cleaned up the main code flow.